test/e2e/test/generation/generation.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package generation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kibanav1 "github.com/elastic/cloud-on-k8s/pkg/apis/kibana/v1"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+)
+
+func getGeneration(obj client.Object, k *test.K8sClient) (int64, error) {
+	if err := k.Client.Get(context.Background(), types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()}, obj); err != nil {
+		return 0, err
+	}
+
+	return obj.GetGeneration(), nil
+}
+
+func getObservedGeneration(obj client.Object, k *test.K8sClient) (int64, error) {
```

Comment:
I like the proposal here: 

https://github.com/elastic/cloud-on-k8s/pull/5383#discussion_r812657424

Reply:
Oh yes, I'd missed that.  That certainly makes this much cleaner.